plugins/net_plugin/include/eosio/net_plugin/auto_bp_peering.hpp

```diff
@@ -167,37 +130,30 @@ class bp_connection_manager {
              established_client_connection(new_connection) && num_established_clients() > self()->connections.get_max_client_count();
    }
 
-   // Only called from main thread
-   neighbor_finder_type neighbor_finder(const std::vector<chain::producer_authority>& schedule) const {
-      return neighbor_finder_type(config, schedule);
-   }
-
    // Only called from main thread
    void on_pending_schedule(const chain::producer_authority_schedule& schedule) {
       if (auto_bp_peering_enabled() && self()->in_sync()) {
          if (schedule.producers.size()) {
             if (pending_schedule_version != schedule.version) {
-               /// establish connection to the BPs within our pending scheduling proximity
+               /// establish connection to our configured BPs, resolve_and_connect ignored if already connected
 
                fc_dlog(self()->get_logger(), "pending producer schedule switches from version ${old} to ${new}",
                        ("old", pending_schedule_version)("new", schedule.version));
 
-               auto finder                       = neighbor_finder(schedule.producers);
-               auto pending_downstream_neighbors = finder.downstream_neighbors();
+               auto pending_connections = configured_bp_accounts(config, schedule.producers);
 
-               fc_dlog(self()->get_logger(), "pending_downstream_neighbors: ${pending_downstream_neighbors}",
-                       ("pending_downstream_neighbors", to_string(pending_downstream_neighbors)));
-               for (auto neighbor : pending_downstream_neighbors) { self()->connections.resolve_and_connect(config.bp_peer_addresses[neighbor], self()->get_first_p2p_address() ); }
+               fc_dlog(self()->get_logger(), "pending_connections: ${c}", ("c", to_string(pending_connections)));
+               for (auto i : pending_connections) {
+                  self()->connections.resolve_and_connect(config.bp_peer_addresses[i], self()->get_first_p2p_address() );
```

Comment:
If `resolve_and_connect` fails, do we still keep the peer in `pending_connections` (and then `pending_configured_bps`)?

Reply:
Yes. And the `net_plugin` will retry to connect to it every `connection-cleanup-period`.